### PR TITLE
Use steed.map.

### LINKED ISSUE #13: Handle connection errors
I think if a connection error is received, it should be the test over, the error should be emitted and existing connections torn down. [This code](https://github.com/nearform/anger/blob/master/anger.js#L39-L42) never runs though because no errors are propagated from `steed.each()`. Maybe it should be `steed.map()` instead?

An alternative to stopping the test is to allow the test to continue, recording the error event and letting it time out when ultimately all the expected responses aren't received. In the end reporting the number of connection errors and the number of subscription errors.

### PR DESCRIPTION
**Pull request by mcollina**
Fixes #13.

@mtharrison @thekemkid please review.

### CODE REVIEW

File: test/failed-auth.test.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        })
+        return uid
+      }
+    })
+
+    instance.on('error', (err) => {
+      t.ok(err, 'error happened')
+    })
+
+    instance.on('end', (result) => {
```
Comment (thekemkid) [inline test/failed-auth.test.js]: does the linter not complain about result being unused?

**Comment by thekemkid** [comment]
single style nit/q, but othwise lgtm

**Comment by mtharrison** [comment]
Looks good. At the moment if some clients fail and some succeed to connect, anger will hang. This won't be a problem when we have a timeout, but what do you think about recording number of failed connections/subscriptions and reporting them on finish?

**Comment by thekemkid** [comment]
That should be a form of error recorded for sure. We should also default `expectedResponses` to be the number of _successfully_ connected clients * the num requests, otherwise the expected response's will be out of whack, causing hangs, afaik

**Comment by mcollina** [comment]
@mtharrison I think that if a client fails to connect and subscribe, we should error straight away. Maybe we could implement exponential backoff: creating a lot of clients at the same time might fail for network reason.

File: test/failed-auth.test.js (hunk 1 above)
Comment (mcollina) [inline test/failed-auth.test.js]: seems not. Maybe it's a bug in eslint with arrow functions! Good catch!

**Comment by mcollina** [comment]
@mtharrison please test this now!

**Pull request merged**